src/containers/Protocol.js

```diff
@@ -12,22 +12,23 @@ class Protocol extends Component {
 
   componentWillMount() {
     if (!this.props.protocol.protocolLoaded) {
-      this.props.loadProtocol()
+      this.props.loadProtocol(this.props.stageId);
     }
   }
 
   render() {
     return (
       <div className='protocol'>
-        <Stage />
+        <Stage id={this.props.stageId} />
       </div>
     );
   }
 }
 
-function mapStateToProps(state) {
+function mapStateToProps(state, ownProps) {
   return {
-    protocol: state.protocol
+    protocol: state.protocol,
+    stageId: ownProps.params.id
```

Comment:
This is how you get the info from the router initially. But if we move away from using the router this is not necessary either.